src/components/v5/common/ActionSidebar/partials/Motions/steps/StakingStep/hooks.tsx

```diff
@@ -0,0 +1,162 @@
+        domainId: Number(nativeMotionDomainId),
+        rootHash,
+      },
+    },
+  });
+
+  const userReputation = data?.getUserReputation;
+  /* User cannot stake more than their reputation in tokens. */
+  const userMaxStake = BigNumber.from(userReputation ?? '0');
+
+  const enoughReputationToStakeMinimum =
+    userMaxStake.gt(0) && userMaxStake.gte(userMinStake);
+
+  return {
+    isLoading:
+      userLoading ||
+      walletConnecting ||
+      loadingUserTokenBalance ||
+      loadingReputation,
+    enoughTokensToStakeMinimum,
+    enoughReputationToStakeMinimum,
+    userActivatedTokens,
+  };
+};
+
+export const useStakingInformation = (
+  usersStakes: ColonyMotionFragment['usersStakes'],
+  tokenDecimals: number,
+  tokenSymbol: string,
+): {
+  votesFor: UserInfoListItem[];
+  votesAgainst: UserInfoListItem[];
+  isLoading?: boolean;
+} => {
+  const { users, loading } = useUsersByAddresses(
+    usersStakes.map((user) => user.address),
+  );
+
+  const votesFor: UserInfoListItem[] =
+    usersStakes?.reduce((result, item) => {
+      if (!item || item.stakes?.raw?.yay === '0') {
+        return result;
+      }
+
+      const userName = users?.length
+        ? users?.find((user) => {
+            const { walletAddress } = user || {};
+            return walletAddress === item.address;
+          })?.profile?.displayName
+        : undefined;
+
+      if (!userName) {
+        return result;
+      }
+
+      // @todo: add sorting
+      return [
+        ...result,
+        {
+          key: item.address,
+          info: formatText(
+            { id: 'motion.staking.staked' },
+            {
+              value: (
+                <Numeral
+                  value={item.stakes?.raw?.yay}
+                  decimals={tokenDecimals}
+                  suffix={tokenSymbol}
+                />
+              ),
+            },
+          ),
+          userProps: {
+            userName,
+          },
+        },
+      ];
+    }, []) || [];
+
+  const votesAgainst: UserInfoListItem[] =
+    usersStakes?.reduce((result, item) => {
+      if (!item || item.stakes?.raw?.nay === '0') {
+        return result;
+      }
+
+      const userName = users?.length
+        ? users?.find((user) => {
+            const { walletAddress } = user || {};
+            return walletAddress === item.address;
+          })?.profile?.displayName
+        : undefined;
+
+      if (!userName) {
+        return result;
+      }
+
+      return [
+        ...result,
+        {
+          key: item.address,
+          info: formatText(
+            { id: 'motion.staking.staked' },
+            {
+              value: (
+                <Numeral
+                  value={item.stakes?.raw?.nay}
+                  decimals={tokenDecimals}
+                  suffix={tokenSymbol}
+                />
+              ),
+            },
+          ),
+          userProps: {
+            userName,
+          },
+        },
+      ];
+    }, []) || [];
```

Comment:
Tiis reduce seems to be the same as the one above except yay/nay, maybe you can make this code common?